Modix.Bot/Modules/UserInfoModule.cs

```diff
@@ -137,6 +150,44 @@ private async Task AddInfractionsToEmbed(ulong userId, StringBuilder builder)
             builder.AppendLine(FormatUtilities.FormatInfractionCounts(counts));
         }
 
+        private async Task AddParticipationToEmbed(ulong userId, StringBuilder builder)
+        {
+            var messages = await MessageRepository.GetRecentUserMessagesAsync(Context.Guild.Id, userId, TimeSpan.FromDays(30));
+
+            var lastWeek = DateTimeOffset.UtcNow - TimeSpan.FromDays(7);
+            var pastWeekCount = messages.Where(x => x.Timestamp >= lastWeek).Count();
+
+            builder.AppendLine();
+            builder.AppendLine("**\u276F Guild Participation**");
+            builder.AppendLine("Last 7 days: " + pastWeekCount + " messages");
+            builder.AppendLine("Last 30 days: " + messages.Count + " messages");
+
+            if (messages.Count > 0)
+            {
+                try
+                {
+                    var channels = messages.GroupBy(x => x.ChannelId)
+                        .Select(x => (ChannelId: x.Key, MessageCount: x.Count()))
+                        .OrderByDescending(x => x.MessageCount)
+                        .ToArray();
+
+                    foreach (var (channelId, messageCount) in channels)
```

Comment:
Will this just append a line for every public channel? I'm not seeing a way to break out of this loop

Reply:
Yep, it would. I've amended the commit to only append the first public channel.